framework/db/src/com/cloud/utils/db/GenericDaoBase.java

```diff
@@ -969,7 +969,12 @@ public T findByUuidIncludingRemoved(final String uuid) {
     @Override
     @DB()
     public T findByIdIncludingRemoved(ID id) {
-        return findById(id, true, null);
+        if (_cache != null) {
+            final Element element = _cache.get(id);
+            return element == null ? findById(id, true, null) : (T)element.getObjectValue();
+        } else {
```

Comment:
@marcaurele good point so to address your concern:
```
T result = null
if (_cache != null) {
    final Element element = _cache.get(id);
    if (element != null) {
        result = (T)element.getObjectValue();
    }
} else {
    result =findById(id, true, null);
}
return result;
```
 this also implements the good practice of having only one return in the method. I am not hellbound on these changes however. As the method is so small it would be merely cosmetic.